presentation/components/src/presentation-components/tree/controlled/PresentationTreeNodeRenderer.tsx

```diff
@@ -26,21 +28,30 @@ export function PresentationTreeRenderer(props: TreeRendererProps) {
  * @alpha
  */
 export interface PresentationTreeNodeRendererProps extends TreeNodeRendererProps {
-  nodeRenderer?: (props: TreeNodeRendererProps) => React.ReactNode;
+  onFilter?: (item: TreeNodeItem) => void;
 }
 
 /**
  * @alpha
  */
 export function PresentationTreeNodeRenderer(props: PresentationTreeNodeRendererProps) {
-  const { nodeRenderer, ...restProps } = props;
+  const { onFilter, className, ...restProps } = props;
   if (restProps.node.item.extendedData !== undefined && restProps.node.item.extendedData.tooManyChildren === true)
     return <TooManyChildNodeRenderer
       nodeDepth={restProps.node.depth}
     />;
-  return nodeRenderer ? <>{nodeRenderer(restProps)}</> : <TreeNodeRenderer
-    {...restProps}
-  />;
+  return <TreeNodeRenderer
+    {...restProps }
+    className={classnames("presentation-node", className)}
+  >
+    {onFilter && <IconButton className="presentation-filter-action-button"
+      styleType="borderless"
+      onClick={(e) => {
+        onFilter?.(restProps.node.item);
```

Comment:
No need to use optional chaining operator `?.` if you are rendering button only when `onFilter` is defined